### PR TITLE
Fix flakiness in opentelemetry test

### PR DESCRIPTION
**Pull request by sfc-gh-yuwang**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-NNNNNNN

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

**Comment by sfc-gh-yuwang** [comment]
All the unit test is covered by integ test so it is safe to move all of them to integ test. The combine coverage test is a proof.

File: tests/unit/test_open_telemetry.py [hunk 1]
```diff
@@ -109,366 +98,6 @@ def minus_udf(x: int, y: int) -> int:
     dict_exporter.clear()
 
 
-def test_register_udaf_from_file(dict_exporter):
-    test_file = os.path.normpath(
-        os.path.join(
-            os.path.dirname(__file__),
```
Comment (sfc-gh-yzou) [inline tests/unit/test_open_telemetry.py:115]: it seems we removed all those tests? do we have a plan to add it back? or those tests are not necessary? 
Also, curious, what is causing the flakyness?

Comment (sfc-gh-yuwang) [reply tests/unit/test_open_telemetry.py:115]: I put a comment above, all these tests are covered in integ tests, so it should be safe to remove them

Comment (sfc-gh-yuwang) [reply tests/unit/test_open_telemetry.py:115]: The flakiness is caused by race condition I believe. When testing opentelemetry, I had to initialize a exporter. The exporter is like a global resource that cannot be re-initialized. Which makes all the tests accessing one resource. I am still looking if there is a better way to solve this issue

Comment (sfc-gh-yzou) [reply tests/unit/test_open_telemetry.py:115]: I see. could you update the pr description with what are we doing in this pr, and why the test could be removed safely?

**Review by sfc-gh-yzou — APPROVED** [review]

File: tests/conftest.py [hunk 2]
```diff
@@ -95,3 +95,42 @@ def clear_warning_dict():
     # clear the warning dict so that warnings from one test don't affect
     # warnings from other tests.
     warning_dict.clear()
+
+
+def attr_to_dict(span):
```
Comment (sfc-gh-aalam) [inline tests/conftest.py:100]: these functions look like they belong to test utils instead of conftest

**Review by sfc-gh-aalam — COMMENTED** [review]
I see that you are using new functions to check for correctness don't see anything specifically handling the race condition you mentioned. Can you point me to that?

File: tests/utils.py [hunk 3]
```diff
@@ -1453,3 +1453,42 @@ class TypeMap(NamedTuple):
...
+                return False
+            else:
+                continue
+        if expected_answer[answer_name] != result[answer_name]:
+            return False
+    return True
+
+
+def check_answers(results, expected_answer):
+    results = span_extractor(results)
```
Comment (sfc-gh-yzou) [inline tests/utils.py:1489]: that seems used very specific for span check, can you rename the function to something like check_tracing_result or check_telemetry_result to be more clear, and add some comment for the function to describe what it is used for?

File: tests/utils.py [hunk 4]
```diff
@@ -1453,3 +1453,42 @@ class TypeMap(NamedTuple):
...
     TypeMap("geometry", "geometry", GeometryType()),
 ]
+
+
+def attr_to_dict(span):
+    dict_attr = {}
+    for name in span.attributes:
+        dict_attr[name] = span.attributes[name]
+    dict_attr["code.filepath"] = os.path.basename(dict_attr["code.filepath"])
+    dict_attr["status_code"] = span.status.status_code
```
Comment (sfc-gh-yzou) [inline tests/utils.py:1463]: can we add type hint for those functions?

File: tests/utils.py [hunk 5]
```diff
@@ -1453,3 +1453,42 @@ class TypeMap(NamedTuple):
...
+
+def span_extractor(dict_exporter):
+    spans = []
+    raw_spans = dict_exporter.get_finished_spans()
+    for span in raw_spans:
+        spans.append((span.name, attr_to_dict(span), span))
+    return spans
+
+
+def check_single_answer(result, expected_answer):
```
Comment (sfc-gh-yzou) [inline tests/utils.py:1476]: similar, check_single_answer -> check_single_span_result etc

Comment (sfc-gh-yzou) [reply tests/utils.py:1476]: if nobody else is calling check_single_answer, there is probably no need to have a separate function for this

**Review by sfc-gh-yzou — COMMENTED** [review]
Hi @sfc-gh-yuwang i left some minor comment about code part, but i don't think i have the expertise to review the fix for flakyness of the test. i will let @sfc-gh-aalam or someone else who knows this part better to make the decision

File: tests/utils.py (hunk 5 above)
Comment (sfc-gh-yuwang) [reply tests/utils.py:1476]: I will add docstrings and type hints.
This is a helper function for check_answer, it is not meant to directly be used in tests, so I have to keep this

**Review by sfc-gh-jdu — APPROVED** [review]

**Review by sfc-gh-aalam — APPROVED** [review]

**Pull request merged**